src/trading/NewOrders.sol

```diff
@@ -0,0 +1,342 @@
+/**
+ * Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+ */
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/NewOrdersFetcher.sol';
+
+
+/**
+ * @title Orders
+ * @dev Storage of all data associated with orders
+ */
+// FIXME: This was named NewOrders to prevent naming conflicts with orders.se. Should be renamed to Orders.
+contract NewOrders is Controlled {
+    using SafeMathUint256 for uint256;
+
+    event CancelOrder(address indexed market, address indexed sender, uint256 fxpPrice, uint256 fxpAmount, bytes20 orderId, uint8 outcome, uint256 orderType, uint256 cashRefund, uint256 sharesRefund);
+    event CompleteSets(address indexed sender, address indexed market, uint256 indexed orderType, uint256 fxpAmount, uint256 numOutcomes, uint256 marketCreatorFee, uint256 reportingFee);
+    event MakeOrder(address indexed market, address indexed sender, uint256 indexed orderType, uint256 fxpPrice, uint256 fxpAmount, uint8 outcome, bytes20 orderId, uint256 fxpMoneyEscrowed, uint256 fxpSharesEscrowed, uint256 tradeGroupId);
+    event TakeOrder(address indexed market, uint256 indexed outcome, uint256 indexed orderType, bytes20 orderId, uint256 price, address maker, address taker, uint256 makerShares, uint256 makerTokens, uint256 takerShares, uint256 takerTokens, uint256 tradeGroupId);
+
+    struct Order {
+        uint256 fxpAmount;
+        uint256 fxpPrice;
+        address owner;
+        uint256 fxpSharesEscrowed;
+        uint256 fxpMoneyEscrowed;
+        bytes20 betterOrderId;
+        bytes20 worseOrderId;
+        uint256 gasPrice; // FIXME: Remove deprecated gasPrice
+    }
+
+    struct MarketOrders {
+        uint256 volume;
+        uint256[] prices;
+    }
+
+    // Trade types
+    // FIXME: Ideally, this should probably be an enum
+    uint8 private constant BID = 1;
```

Comment:
I suppose this enum should probably be put in a separate library, since most of the trading contracts use these constants.